### ISSUE TITLE
Not compatible with NUnit 3.10.1

### ISSUE THREAD

**Issue by MichaelMayrhofer** [AkkaNetContrib/Akka.TestKit.NUnit#37]
```
        [Test]
        public void TestMethod1()
        {
            int n = 0;
            AwaitAssert(() =>
            {
                n++;
                Assert.That(n > 1);
            });
        }
```
...works fine with
 * Akka.TestKit.NUnit3 v1.2.1
 * Akka.TestKit v1.3.2
 * NUnit v3.9.0

Updating Akka.TestKit.NUnit3 to latest version v1.3.8 updates NUnit to version v3.10.1. Now the test will be marked as "failed".

This is because NUnit v3.10.1 changed the behaviour of AssertionException handling (see https://github.com/nunit/nunit/issues/2758).

**Comment by Danthar** [comment]
Yeah this is a breaking change on the part of NUnit. Our AwaitAssert still works as it should. As in, it keeps trying untill the assertion no longer throws, or the timeout duration is reached.
However when it returns due to the assertion no longer failing. The Test still reports as failed because NUnit keeps a hold of the previous errors.

It seems like NUnit no longer reports the last assertion state in this case.

**Comment by Danthar** [comment]
You can easily test it by checking N after the AwaitAssert. You will see its 2 and thus the Assert.That would have reported true. And no exception occurs (because your hitting N after the AwaitAssert). But NUnit still reports the Test as failed.

**Comment by MichaelMayrhofer** [comment]
I am wondering that there are no tests in the Akka.TestKit.NUnit project which fail after this breaking change in NUnit ;-)

Do you have any ideas how to fix this? I'm not sure if the NUnit guys will revert their changes...

**Comment by Aaronontheweb** [comment]
Resolved in 1.4.39, which we are releasing now
